deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/recurrent/SimpleRnn.java

```diff
@@ -114,12 +126,17 @@ public INDArray rnnActivateUsingStoredState(INDArray input, boolean training, bo
             INDArray dldaCurrent = epsilon.get(all(), all(), point(i));
             INDArray aCurrent = p.getFirst().get(all(), all(), point(i));
             INDArray zCurrent = p.getSecond().get(all(), all(), point(i));
+            INDArray nCurrent = (hasLayerNorm() ? p.getThird().get(all(), all(), point(i)) : null);
+            INDArray rCurrent = (hasLayerNorm() ? p.getFourth().get(all(), all(), point(i)) : null);
             INDArray inCurrent = input.get(all(), all(), point(i));
             INDArray epsOutCurrent = epsOut.get(all(), all(), point(i));
 
             if(dldzNext != null){
                 //Backprop the component of dL/da (for current time step) from the recurrent connections
                 Nd4j.gemm(dldzNext, rw, dldaCurrent, false, true, 1.0, 1.0);
+
+                //Recurrent weight gradients:
+                Nd4j.gemm(aCurrent, dldzNext, rwg, true, false, 1.0, 1.0);
```

Comment:
aCurrent will be null if no layer norm - this will give NPE in that case?

Reply:
it is **rCurrent** that will be null. aCurrent is always defined